### PR TITLE
Enable development openapi spec and customer facing spec

### PR DESCRIPTION
**Pull request by chambridge**
# Overview

This PR is being created to address [RHINENG-8613](https://issues.redhat.com/browse/RHINENG-8613).
* Duplicate existing openapi spec for development use with connexion
* Update cutomer facing openapi spec to remove assignment-rules endpoints
* Add new blueprint for serving customer facing spec
* Alter connexion to load from the development spec
* Update pre-commit check to update the development spec now
* In order to make API visible you will need to update the customer facing spec

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -43,7 +44,7 @@
 
 # temporary replaced for ESSNTL-746 - correct normalization of the bundled spec
 # SPECIFICATION_FILE = join(SPECIFICATION_DIR, "api.spec.yaml")
-SPECIFICATION_FILE = join(SPECIFICATION_DIR, "openapi.json")
+SPECIFICATION_FILE = join(SPECIFICATION_DIR, "openapi.dev.json")
```
Comment (jpramos123) [inline app/__init__.py:47]: @chambridge The idea is to use `SPEFICITAION_DIR` variable to shift between `dev` and `prod` files?

**Review by jpramos123 — APPROVED** [review]
LGTM

File: app/__init__.py (hunk 1 above)
Comment (chambridge) [reply app/__init__.py:47]: The idea is to allow Connexion to load and configure the app using the Open API dev spec, but it wont be exposed at the openapi.json endpoint that will be served by another endpoint.

**Review by strider — APPROVED** [review]
LGTM

**Pull request merged**